packages/expo-background-task/ios/BackgroundTaskConsumer.swift

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024-present 650 Industries. All rights reserved.
+import ExpoModulesCore
+
+class BackgroundTaskConsumer: NSObject, EXTaskConsumerInterface {
+  var task: EXTaskInterface?
+  static var numberOfRegisteredTasksOfThisType: Int = 0
+
+  static func supportsLaunchReason(_ launchReason: EXTaskLaunchReason) -> Bool {
+    return launchReason == EXTaskLaunchReasonBackgroundTask
+  }
+
+  func taskType() -> String {
+    return "backgroundTask"
+  }
+
+  func normalizeTaskResult(_ result: Any?) -> UInt {
+    guard let result = result as? Int else {
+      return UIBackgroundFetchResult.noData.rawValue
+    }
+
+    switch result {
+    case BackgroundTaskResult.success.rawValue:
+      return UIBackgroundFetchResult.newData.rawValue
+    case BackgroundTaskResult.failed.rawValue:
+      return UIBackgroundFetchResult.failed.rawValue
+    default:
+      return UIBackgroundFetchResult.newData.rawValue
+    }
+  }
+
+  func didBecomeReadyToExecute(withData data: [AnyHashable: Any]?) {
+    // Run on main thread. The task execution needs to be called on the main thread
+    // since it accesses the UIApplication's state
+    EXUtilities.performSynchronously {
```

Comment:
```suggestion
    assert(Thread.isMainThread)
    EXUtilities.performSynchronously {
```

Reply:
I don't think we want to assert here - because we don't know where this will be called from (Task Manager calls this code from the iOS `BGTaskScheduler`'s thread) - what we want is to make sure the `self.task?.execute(withData: data, withError: nil)` line is called on the main thread - if this makes sense.